pkg/storage/replica_command.go

```diff
@@ -3177,11 +3177,11 @@ func (r *Replica) ChangeReplicas(
 		// complete. See #10409.
 		if err := func() error {
 			snap, err := r.GetSnapshot(ctx, snapTypePreemptive)
-			defer r.CloseOutSnap()
-			log.Event(ctx, "generated snapshot")
 			if err != nil {
 				return errors.Wrapf(err, "%s: change replicas failed", r)
```

Comment:
While you're here, this message looks weird. "change replicas failed to generate snapshot: %s"?